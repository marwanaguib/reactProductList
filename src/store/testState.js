export default {
  products: {
    count: 7,
    pageIndex: 0,
    departments: [
      { id: 1, name: "Electronics" },
      { id: 2, name: "Toys" },
      { id: 3, name: "Cloths" },
    ],
    filters: {},
    items: [
      { id: 1, name: "Product1", price: "100", promo: "10%", isActive: 1 },
      { id: 2, name: "Product2", price: "110", promo: "10%", isActive: 0 },
      { id: 3, name: "Product3", price: "80", promo: "10%", isActive: 1 },
      { id: 4, name: "Product4", price: "100", promo: "10%", isActive: 1 },
      { id: 5, name: "Product5", price: "110", promo: "10%", isActive: 1 },
      { id: 6, name: "Product6", price: "100", promo: "10%", isActive: 1 },
      { id: 7, name: "Product7", price: "100", promo: "10%", isActive: 1 },
    ],
  },
  shared: {
    errorMsg: "",
    isLoading: false,
  },
};
