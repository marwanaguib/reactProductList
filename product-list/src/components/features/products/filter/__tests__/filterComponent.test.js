import React from "react";
import { shallow, mount } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import Filter from "../component";
import testState from "../../../../../store/testState";

const props = {
  getDepartments: jest.fn(),
  setFilter: jest.fn(),
  departments: testState.products.departments,
};

describe("products filter", () => {
  it("render Filter to match snapshot", () => {
    const componetWrapper = shallow(<Filter {...props} />);
    expect(componetWrapper).toMatchSnapshot();
  });
  describe("functions called", () => {
    const store = configureStore()(testState);
    const componetWrapper = mount(
      <Provider store={store}>
        <Filter {...props} />
      </Provider>
    );
    it("getDepartments has been called once", () => {
      expect(props.getDepartments).toHaveBeenCalledTimes(1);
      expect(componetWrapper).toMatchSnapshot();
    });

    it("setFilter called when select onChange fire", () => {
      componetWrapper.find("select").simulate("change");
      expect(props.setFilter).toHaveBeenCalledTimes(1);
    });

    it("setFilter called when inputForm submit fire", () => {
      componetWrapper.find("form").at(0).simulate("submit");
      expect(props.setFilter).toHaveBeenCalledTimes(1);
    });
  });
});
