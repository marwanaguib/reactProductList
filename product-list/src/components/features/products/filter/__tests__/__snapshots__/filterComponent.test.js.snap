// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`products filter functions called getDepartments has been called once 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Filter
    departments={
      Array [
        Object {
          "id": 1,
          "name": "Electronics",
        },
        Object {
          "id": 2,
          "name": "Toys",
        },
        Object {
          "id": 3,
          "name": "Cloths",
        },
      ]
    }
    getDepartments={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    setFilter={[MockFunction]}
  >
    <h3>
      Filter Products
    </h3>
    <Select
      elemKey="department"
      func={[MockFunction]}
      key="Departments"
      name="Departments"
      options={
        Array [
          Object {
            "id": 1,
            "name": "Electronics",
          },
          Object {
            "id": 2,
            "name": "Toys",
          },
          Object {
            "id": 3,
            "name": "Cloths",
          },
        ]
      }
      type="select"
    >
      <styled.form>
        <form
          className="sc-AxhCb cNNKgv"
        >
          <select
            onChange={[Function]}
          >
            <option
              value=""
            >
              Select Departments
            </option>
            <option
              key="1"
              value={1}
            >
              Electronics
            </option>
            <option
              key="2"
              value={2}
            >
              Toys
            </option>
            <option
              key="3"
              value={3}
            >
              Cloths
            </option>
          </select>
        </form>
      </styled.form>
    </Select>
    <Input
      elemKey="promo"
      func={[MockFunction]}
      key="Promo Code"
      name="Promo Code"
      type="text"
    >
      <styled.form
        onSubmit={[Function]}
      >
        <form
          className="sc-AxhCb cNNKgv"
          onSubmit={[Function]}
        >
          <input
            placeholder="Promo Code"
            type="text"
          />
          <input
            type="submit"
            value="Search"
          />
        </form>
      </styled.form>
    </Input>
    <Input
      elemKey="search"
      func={[MockFunction]}
      key="Product Name"
      name="Product Name"
      type="text"
    >
      <styled.form
        onSubmit={[Function]}
      >
        <form
          className="sc-AxhCb cNNKgv"
          onSubmit={[Function]}
        >
          <input
            placeholder="Product Name"
            type="text"
          />
          <input
            type="submit"
            value="Search"
          />
        </form>
      </styled.form>
    </Input>
  </Filter>
</Provider>
`;

exports[`products filter render Filter to match snapshot 1`] = `
<Fragment>
  <h3>
    Filter Products
  </h3>
  <Select
    elemKey="department"
    func={[MockFunction]}
    key="Departments"
    name="Departments"
    options={
      Array [
        Object {
          "id": 1,
          "name": "Electronics",
        },
        Object {
          "id": 2,
          "name": "Toys",
        },
        Object {
          "id": 3,
          "name": "Cloths",
        },
      ]
    }
    type="select"
  />
  <Input
    elemKey="promo"
    func={[MockFunction]}
    key="Promo Code"
    name="Promo Code"
    type="text"
  />
  <Input
    elemKey="search"
    func={[MockFunction]}
    key="Product Name"
    name="Product Name"
    type="text"
  />
</Fragment>
`;
